packages/mdx/test/compile.js

```diff
@@ -859,9 +859,9 @@ test('jsx', async () => {
       '/*@jsxRuntime automatic @jsxImportSource react*/',
       'function _createMdxContent(props) {',
       '  const _components = Object.assign({',
-      '    "a-b": "a-b"',
+      '    a_b: "a-b"',
       '  }, props.components);',
-      '  return <>{<_components.a-b></_components.a-b>}</>;',
```

Comment:
This test asserts that the output is _invalid_ JSX! Converting dashes `-` to underscores `_` should resolve this issue for custom elements 👍